File: src/CouchChefFrontend/src/app/entities/recipe.dto.ts
import { GetImageDTO, PostImageDTO } from "./image.dto";

export interface PostRecipeDTO {
    id: number,
    cuisineId: number,
    name: string,
    prepareTime: Date,
    totalTime: Date,
    servings: number,
    directions: string,
    imageId: number,
    postImageDTO: PostImageDTO,
    categoryIds: number[],
    postRecipeIngredientDetailDTOs: PostRecipeIngredientDetailDTO[]
}

export interface PostRecipeIngredientDetailDTO {
    ingredientId: number,
    isTagged: boolean,
    weightInGrams: number
}

export interface RecipeDTO {
    id: number,
    name: string,
    prepareTime: Date,
    totalTime: Date,
    servings: number,
    directions: string,
    protein: number,
    fat: number,
    carbs: number,
    calories: number,
    getImageDTO: GetImageDTO,
    cuisine: string,
    ingredients: string[],
    categories: string[]
}

File: src/CouchChefBackend/CouchChefBLL/DTOs/RecipeDTO.cs
using CouchChefBLL.DTOs.Get;

namespace CouchChefBLL.DTOs;

public class RecipeDTO
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public TimeOnly PrepareTime { get; set; }
    public TimeOnly TotalTime { get; set; }
    public int Servings { get; set; }
    public required string Directions { get; set; }
    public float Protein { get; set; }
    public float Fat { get; set; }
    public float Carbs { get; set; }
    public float Calories { get; set; }

    public required GetImageDTO GetImageDTO { get; set; }
    public required string Cuisine { get; set; }
    public required List<string> Ingredients { get; set; }
    public required List<string> Categories { get; set; }
}
